portfolio/src/main/webapp/script.js

```diff
@@ -57,29 +57,106 @@ function toggle(project) {
 
-    const commentsListElement = document.getElementById('comment-section');
+    const nps_icon = document.getElementById('nps-icon');
+    if (stats["nps"] > 30) {nps_icon.style.color = "rgba(118,186,90,.85)"}
+    if (stats["nps"] < 0) {nps_icon.style.color = "rgba(220,50,30,.85)"}
+    
+    const commentsListElement = document.getElementById('comments-ul');
     commentsListElement.innerHTML = '';
-    for (i = 0; i < comments.length; i++) {
+    for (i = 1; i < comments.length; i++) {
         commentsListElement.appendChild(
         createListElement(comments[i]));
     }
   });
 }
 
-/** Creates an <li> element containing text. */
-function createListElement(text) {
+function writeStats(stat, name) { 
+    const liElement = document.createElement('li');
+    liElement.setAttribute("id", "stat");
+    if (name === "Total Comments") {
+            liElement.innerHTML = '<i class="fas fa-comments fa-2x" id="comments-icon"></i>'  + 
+                                '<p>' + name + ": " + stat + '</p>';
+    } 
+    if (name === "Average Rating") {
+            liElement.innerHTML = '<i class="fas fa-star-half-alt fa-2x" id="rtg-icon"></i>'  + 
+                                '<p>' + name + ": " + stat + '</p>';
+    } 
+    if (name === "Net Promoter Score") {
+            liElement.innerHTML = '<i class="fas fa-arrow-circle-up fa-2x" id="nps-icon"></i>'  + 
+                                '<p>' + name + ": " + stat + '</p>';
+    } 
+    
+    return liElement;
+}
+
+/* Creates an <li> element containing text. */
+function createListElement(commentObj) {
+  var name = commentObj["name"];
+  var comment = commentObj["comment"];
+  var rating = commentObj["rating"];
+  var id = commentObj["id"];
+  var date = timeConverter(commentObj["timestamp"]);
+
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerHTML = liHTML(name,comment,rating,date, id);
   return liElement;
 }
 
+function liHTML(name,comment,rating, date, id) {
+  var html = '<footer class="post-info">' +
+      '<div class="separated horiz-flex-div" >' +  
+        '<abbr title="' + date + '">' + date + '</abbr>' +
+        '<button onClick="postDeleteComment(' + id + ');" class="inv-btn">' +
+            '<i class="far fa-trash-alt"></i>' +
+        '</button>' + 
+      '</div>' +
+      '<div class="horiz-flex-div">' + 
+        '<address class="author"> <b>' + 'By ' + name + '</b></address>' +
+        '<address class="rtg">' + 'Rating: ' + rating + '</address>' +
+      '</div>' +
+      '</footer>' +
+      '<div class="comment-text-div">' + '<p>' + comment + '</p>' + '</div>';
+
+  return html;
+}
+
+function postDeleteComment(id) {
+    var params = new URLSearchParams();
+    params.append('id', id);
+    let response = fetch('/delete-comment', {method: 'POST', body: params}).then(getComments());
+}
+
+function timeConverter(UNIX_timestamp){
+  var a = new Date(UNIX_timestamp);
```

Comment:
it seems like there should be a library for converting date to string with a custom format? maybe check with brooke or the other interns.

Reply:
I looked on stackoverflow and this seemed to be the best way to do it, since you get pretty high customizability

Reply:
Sounds good.